### PR TITLE
BUG: fix write/read roundtrips with empty Table dumped to ECSV

### LINKED ISSUE #13191: Reading an empty table generates a "StopIteration" error
### Description
Using astropy.Table I created an empty table and saved it. When reading this
empty table I encountered a StopIteration error.  

### Expected behavior
I would have expected astropy.Table to read the empty table.

### Actual behavior
A StopIteration error was returned.
The error is generic, so I found it mildly confusing to understand the cause.

### Steps to Reproduce
The minimum working example is provided here:

```
from astropy.table import Table
t = Table()
t.write('x.ecsv')
t_new = Table.read('x.ecsv')
```
**The error returned is:**

---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
/tmp/ipykernel_5419/3218244.py in <module>
      2 t = Table()
      3 t.write('x.ecsv')
----> 4 t_new = Table.read('x.ecsv')

~/anaconda3/lib/python3.9/site-packages/astropy/table/connect.py in __call__(self, *args, **kwargs)
     60         descriptions = kwargs.pop('descriptions', None)
     61 
---> 62         out = self.registry.read(cls, *args, **kwargs)
     63 
     64         # For some readers (e.g., ascii.ecsv), the returned `out` class is not

~/anaconda3/lib/python3.9/site-packages/astropy/io/registry/core.py in read(self, cls, format, cache, *args, **kwargs)
    197 
    198             reader = self.get_reader(format, cls)
--> 199             data = reader(*args, **kwargs)
    200 
    201             if not isinstance(data, cls):

~/anaconda3/lib/python3.9/site-packages/astropy/io/ascii/connect.py in io_read(format, filename, **kwargs)
     16         format = re.sub(r'^ascii\.', '', format)
     17         kwargs['format'] = format
---> 18     return read(filename, **kwargs)
     19 
     20 

~/anaconda3/lib/python3.9/site-packages/astropy/io/ascii/ui.py in read(table, guess, **kwargs)
    374         else:
    375             reader = get_reader(**new_kwargs)
--> 376             dat = reader.read(table)
    377             _read_trace.append({'kwargs': copy.deepcopy(new_kwargs),
    378                                 'Reader': reader.__class__,

~/anaconda3/lib/python3.9/site-packages/astropy/io/ascii/core.py in read(self, table)
   1341 
   1342         # Get the table column definitions
-> 1343         self.header.get_cols(self.lines)
   1344 
   1345         # Make sure columns are valid

~/anaconda3/lib/python3.9/site-packages/astropy/io/ascii/ecsv.py in get_cols(self, lines)
    155         # Read the first non-commented line of table and split to get the CSV
    156         # header column names.  This is essentially what the Basic reader does.
--> 157         header_line = next(super().process_lines(raw_lines))
    158         header_names = next(self.splitter([header_line]))
    159 

StopIteration: 

### System Details
Linux-5.13.0-40-generic-x86_64-with-glibc2.31
Python 3.9.7 (default, Sep 16 2021, 13:09:58) 
[GCC 7.5.0]
Numpy 1.20.3
pyerfa 2.0.0
astropy 5.0.3
Scipy 1.7.1
Matplotlib 3.4.3

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
Fixes #13191

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/io/ascii/tests/test_ecsv.py [hunk 1]
```diff
@@ -156,6 +156,13 @@ def test_write_read_roundtrip():
                 assert np.all(t[name] == t2[name])
 
 
+def test_write_read_roundtrip_empty_table(tmp_path):
+    # see https://github.com/astropy/astropy/issues/13191
+    sfile = tmp_path / "x.ecsv"
+    Table().write(sfile)
+    Table.read(sfile)
```
Comment (mhvk) [inline astropy/io/ascii/tests/test_ecsv.py:163]: Could we add a 
```
t = Table.read(sfile)
assert len(t) == 0
assert len(t.colnames) == 0
```

**Review by mhvk — COMMENTED** [review]
Thanks, that looks good! Just a request to actually check that the re-read table is empty

File: astropy/io/ascii/tests/test_ecsv.py (hunk 1 above)
Comment (neutrinoceros) [reply astropy/io/ascii/tests/test_ecsv.py]: done !

**Review by mhvk — APPROVED** [review]
Thanks!

**Comment by pllim** [comment]
Thanks!

@meeseeksdev backport to v6.0.x

**Pull request merged**